pvl/__init__.py

```diff
@@ -60,20 +62,27 @@ def load(path, parser=None, grammar=None, decoder=None, **kwargs):
 
 def get_text_from(path) -> str:
     try:
-        p = Path(path)
-        return p.read_text()
+        if isinstance(path, str) and path.startswith('http'):
+            with urllib.request.urlopen(path) as f:
+                c = f.read().decode('utf-8').replace('\r', '')
+                return c
+        else:
+            p = Path(path)
+            return p.read_text()
     except UnicodeDecodeError:
         # This may be the result of an ISIS cube file (or anything else)
         # where the first set of bytes might be decodable, but once the
         # image data starts, they won't be, and the above tidy function
         # fails.  So open the file as a bytestream, and read until
         # we can't decode.  We don't want to just run the .read_bytes()
         # method of Path, because this could be a giant file.
-        with open(p, mode='rb') as f:
+        with open(path, mode='rb') as f:
             return decode_by_char(f)
     except TypeError:
+        if isinstance(path, HTTPResponse):
+            return path.read().decode('utf-8')
```

Comment:
Is this needed?  Seems duplicative.  The `path` variable comes in at the top, and you're just trying to get to a point to call `.read()` on `path` which is going to happen if you can get past path.readable() that open file objects have, but that HTTPResponse doesn't.  Maybe we just need to widen the `if path.readable()` to just test that `path` has a `read()` method maybe just via `hasattr()` instead of relying on the `.readable()` method?

Reply:
yeah we can remove it / clean things up a bit